### PR TITLE
use `EnumX`

### PR DESCRIPTION
**Pull request by prbzrg**
I changed `alg_interpretation`  to use EnumX instead of Symbol.

### CODE REVIEW

File: src/StochasticDiffEq.jl [hunk 1]
```diff
@@ -90,6 +91,8 @@ end
     seed_multiplier() = 1
   end
 
+  EnumX.@enumx AlgorithmInterpretation Ito Stratonovich
```
Comment (ChrisRackauckas) [inline src/StochasticDiffEq.jl:94]: This should be defined in SciMLBase.

Comment (prbzrg) [reply src/StochasticDiffEq.jl]: https://github.com/SciML/SciMLBase.jl/pull/777

**Comment by prbzrg** [comment]
needs https://github.com/SciML/SciMLBase.jl/pull/777

File: Project.toml [hunk 2]
```diff
@@ -37,6 +38,7 @@ DataStructures = "0.18"
 DiffEqBase = "6.154"
 DiffEqNoiseProcess = "5.13"
 DocStringExtensions = "0.8, 0.9"
+EnumX = "1"
```
Comment (ChrisRackauckas) [inline Project.toml:41]:
```suggestion
```

File: Project.toml [hunk 3]
```diff
@@ -10,6 +10,7 @@ DataStructures = "864edb3b-99cc-5e75-8d2d-829cb0a9cfe8"
 DiffEqBase = "2b5f629d-d688-5b77-993f-72d75c75574e"
 DiffEqNoiseProcess = "77a26b50-5914-5dd7-bc55-306e6241c503"
 DocStringExtensions = "ffbed154-4ef7-542d-bbb7-c09d3a79fcae"
+EnumX = "4e289a0a-7415-4d19-859d-a7e5c4648b56"
```
Comment (ChrisRackauckas) [inline Project.toml:13]:
```suggestion
```

File: src/StochasticDiffEq.jl [hunk 4]
```diff
@@ -73,6 +73,7 @@ else
 end
 
   import SciMLBase
+  import EnumX
```
Comment (ChrisRackauckas) [inline src/StochasticDiffEq.jl:76]:
```suggestion
```

**Pull request merged**